### PR TITLE
fix: disable apiserver insecure-port

### PR DESCRIPTION
**Pull request by sozercan**
**Reason for Change**:

Disables deprecated apiserver insecure-port

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by ritazh** [comment]
cc @robbiezhang @palma21 FYI

**Comment by jackfrancis** [comment]
One symptom from kube-addon-manager:

```
$ docker logs 23776420d2d0
INFO: == Generated kubectl prune whitelist flags: --prune-whitelist core/v1/ConfigMap --prune-whitelist core/v1/Endpoints --prune-whitelist core/v1/Namespace --prune-whitelist core/v1/PersistentVolumeClaim --prune-whitelist core/v1/PersistentVolume --prune-whitelist core/v1/Pod --prune-whitelist core/v1/ReplicationController --prune-whitelist core/v1/Secret --prune-whitelist core/v1/Service --prune-whitelist batch/v1/Job --prune-whitelist batch/v1beta1/CronJob --prune-whitelist apps/v1/DaemonSet --prune-whitelist apps/v1/Deployment --prune-whitelist apps/v1/ReplicaSet --prune-whitelist apps/v1/StatefulSet --prune-whitelist extensions/v1beta1/Ingress  ==
INFO: == Kubernetes addon manager started at 2020-06-05T22:20:59+00:00 with ADDON_CHECK_INTERVAL_SEC=60 ==
The connection to the server localhost:8080 was refused - did you specify the right host or port?
WRN: == Error getting default service account, retry in 0.5 second ==
The connection to the server localhost:8080 was refused - did you specify the right host or port?
WRN: == Error getting default service account, retry in 0.5 second ==
The connection to the server localhost:8080 was refused - did you specify the right host or port?
WRN: == Error getting default service account, retry in 0.5 second ==
The connection to the server localhost:8080 was refused - did you specify the right host or port?
WRN: == Error getting default service account, retry in 0.5 second ==
```

File: parts/k8s/manifests/kubernetesmaster-kube-addon-manager.yaml [hunk 1]
```diff
@@ -24,11 +27,23 @@ spec:
...
       path: /etc/kubernetes/addons/init
   - name: msi
     hostPath:
       path: /var/lib/waagent/ManagedIdentity-Settings
+  - name: var-lib-kubelet
+    hostPath:
+      path: /var/lib/kubelet
+  - name: etc-kubernetes
+    hostPath:
+      path: /etc/kubernetes
```
Comment (jackfrancis) [inline parts/k8s/manifests/kubernetesmaster-kube-addon-manager.yaml:48]: You might need `type: Directory` here, unless it's smart enough to figure that out

Comment (sozercan) [reply parts/k8s/manifests/kubernetesmaster-kube-addon-manager.yaml:48]: I was just looking at the CCM yaml, don't think we need it https://github.com/Azure/aks-engine/blob/master/parts/k8s/manifests/kubernetesmaster-cloud-controller-manager.yaml

**Pull request merged**